### PR TITLE
Web3 artifacts wrapper

### PR DESCRIPTION
**Pull request by patjanawat**
- Init web3.js (**Resolved**)
- Init and return the smart contracts (**Resolved**)
- Check for available accounts. No need thanks to Metamask
- Create needed helper functions: No need thanks to Metamask

Closes https://github.com/mesg-foundation/atd/issues/35

### CODE REVIEW

File: frontend/artifacts/index.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import Web3 from 'web3'
+const config = require(`./config${ process.env.isDev ? '.dev' : '' }.json`)
+import * as abi from './abi'
+
+let provider = config.ethereum_node
+if (typeof window.ethereum !== 'undefined' || (typeof window.web3 !== 'undefined')) {
+  // Web3 browser user detected. You can now use the provider.
+  provider = window['ethereum'] || window.web3.currentProvider
+  console.log('Metamask detected')
```
Comment (NicolasMahe) [inline frontend/artifacts/index.js]:
```suggestion
  console.log('Web3 browser user detected')
```

**Comment by patjanawat** [comment]
> I have an error when I load the website https://deploy-preview-43--mesg-atd.netlify.com/
> 
> `Cannot read property 'toNumber' of null`

I will fix it.

**Comment by patjanawat** [comment]
> I have an error when I load the website https://deploy-preview-43--mesg-atd.netlify.com/
> 
> `Cannot read property 'toNumber' of null`

Resolved

File: frontend/artifacts/index.js [hunk 2]
```diff
@@ -5,9 +5,10 @@ import * as abi from './abi'
 let provider = config.ethereum_node
 if (typeof window.ethereum !== 'undefined' || (typeof window.web3 !== 'undefined')) {
   // Web3 browser user detected. You can now use the provider.
-  provider = window['ethereum'] || window.web3.currentProvider
+  provider = (window['ethereum'] || window.web3.currentProvider).selectedAddress ? window['ethereum'] || window.web3.currentProvider : config.ethereum_node
```
Comment (NicolasMahe) [inline frontend/artifacts/index.js]: What is `selectedAddress` and why it is used here?
What I see when google this is actually Metamask that is in privacy mode and the dashboard need to call `ethereum.enable()`.
source: https://github.com/MetaMask/metamask-extension/issues/6491
Can you elaborate on why this fix is done like this please?

**Comment by NicolasMahe** [comment]
> I have an error when I load the website https://deploy-preview-43--mesg-atd.netlify.com/
> 
> `Cannot read property 'toNumber' of null`

This error is related to the selected network in metamask. There is already a issue created: https://github.com/mesg-foundation/atd/issues/44

@antho1404 please try again and make sure to select the mainnet in metamask.

File: frontend/artifacts/index.js (hunk 2 above)
Comment (patjanawat) [reply frontend/artifacts/index.js]: Ok, I see,  I confused about how to use Metamask. I will revert the last commit and fix it on issue [#44](https://github.com/mesg-foundation/atd/issues/44)

**Comment by patjanawat** [comment]
> Please revert last commit ([f63cacd](https://github.com/mesg-foundation/atd/commit/f63cacd00352c06dfeddd721da2c78882113cf90)) as it is not related to the error

Reverted.

**Pull request merged**